src/main/java/com/epam/aidial/core/controller/DeleteFileController.java

```diff
@@ -31,27 +35,29 @@ protected Future<?> handle(ResourceDescription resource, boolean hasWriteAccess)
             return context.respond(HttpStatus.BAD_REQUEST, "Can't delete a folder");
         }
 
-        String absoluteFilePath = resource.getAbsoluteFilePath();
+        EtagHeader etag = EtagHeader.fromRequest(context.getRequest());
 
-        BlobStorage storage = proxy.getStorage();
         Future<Void> result = proxy.getVertx().executeBlocking(() -> {
             String bucketName = resource.getBucketName();
             String bucketLocation = resource.getBucketLocation();
-            try {
-                return lockService.underBucketLock(bucketLocation, () -> {
-                    invitationService.cleanUpResourceLink(bucketName, bucketLocation, resource);
-                    shareService.revokeSharedResource(bucketName, bucketLocation, resource);
-                    storage.delete(absoluteFilePath);
-                    return null;
-                });
-            } catch (Exception ex) {
-                log.error("Failed to delete file  %s/%s".formatted(bucketName, resource.getOriginalPath()), ex);
-                throw new RuntimeException(ex);
-            }
+            return lockService.underBucketLock(bucketLocation, () -> {
+                fileService.deleteFile(resource, etag);
```

Comment:
Why did you change the order? The subsequent operations can fail - a file is deleted, but other links are not. The previous version is a bit  safer than this one.

Reply:
Moved down.